### ISSUE TITLE
Cannot override method with lambda (introduced in 1.1.334)

### ISSUE THREAD

**Issue by last-partizan** [microsoft/pyright#6308]
**Describe the bug**
I was using lambdas instead of some methods in django, and now it does not work.

When using normal `def`, same code is working fine.

**Code or Screenshots**
Here's example:

```python
class Admin:
    def has_delete_permission(self, request, obj=None) -> bool:
        ...


class CustomAdmin(Admin):
    has_delete_permission = lambda *args, **kwargs: False
```

Produces following error:

```
  /tmp/admin.py:7:29 - error: Expression of type "(*args: Self@Admin, **kwargs: Unknown) -> Literal[False]" cannot be assigned to declared type "(self: Self@Admin, request: Unknown, obj: Unknown | None = None) -> bool"
    Type "(*args: Self@Admin, **kwargs: Unknown) -> Literal[False]" cannot be assigned to type "(self: Self@Admin, request: Unknown, obj: Unknown | None = None) -> bool"
      Parameter 3: type "Unknown | None" cannot be assigned to type "Self@Admin"
        Type "Unknown | None" cannot be assigned to type "Admin"
          "None" is incompatible with "Admin" (reportGeneralTypeIssues)
1 error, 0 warnings, 0 informations
```

**VS Code extension or command-line**

* pyright 1.1.334

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.335, which I just published. It will also be included in a future release of pylance.
